File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse127.py

import sys
from io import StringIO

MAX_USERNAME_LENGTH = 100

def sanitize_input(input_str):
    """Sanitize input to prevent injection attacks"""
    if not input_str:
        return ""
    
    html_escape_table = {
        '<': '&lt;',
        '>': '&gt;',
        '&': '&amp;',
        '"': '&quot;',
        "'": '&#x27;',
        '/': '&#x2F;'
    }
    
    sanitized = []
    for char in input_str:
        if char in html_escape_table:
            sanitized.append(html_escape_table[char])
        elif 32 <= ord(char) <= 126:
            sanitized.append(char)
    
    return ''.join(sanitized)

def http_hello_page(request, response):
    """Process HTTP request and generate hello response"""
    try:
        username = request.readline().strip()
        
        # Input validation and sanitization
        if not username:
            username = "Guest"
        else:
            if len(username) > MAX_USERNAME_LENGTH:
                username = username[:MAX_USERNAME_LENGTH]
            username = sanitize_input(username)
        
        response_message = f"Hello {username}"
        response.write(response_message + "\\n")
        response.flush()
        
        return response_message
    except Exception as e:
        error_msg = "Error processing request"
        response.write(error_msg + "\\n")
        return error_msg

def main():
    test_cases = [
        "Alice",
        "Bob<script>alert('xss')</script>",
        "Charlie&Drop Table",
        "",
        "A" * 150
    ]
    
    for i, test_input in enumerate(test_cases, 1):
        request = StringIO(test_input + "\\n")
        response = StringIO()
        
        result = http_hello_page(request, response)
        print(f"Test {i}: {result}")

if __name__ == "__main__":
    main()

File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse127.java

import java.io.*;
import java.util.Scanner;

public class Lse127 {
    private static final int MAX_USERNAME_LENGTH = 100;
    
    public static String httpHelloPage(InputStream request, OutputStream response) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(request));
        PrintWriter writer = new PrintWriter(new OutputStreamWriter(response), true);
        
        try {
            String username = reader.readLine();
            
            // Input validation and sanitization
            if (username == null || username.trim().isEmpty()) {
                username = "Guest";
            } else {
                username = username.trim();
                if (username.length() > MAX_USERNAME_LENGTH) {
                    username = username.substring(0, MAX_USERNAME_LENGTH);
                }
                username = sanitizeInput(username);
            }
            
            String responseMessage = "Hello " + username;
            writer.println(responseMessage);
            writer.flush();
            
            return responseMessage;
        } finally {
            writer.close();
            reader.close();
        }
    }
    
    private static String sanitizeInput(String input) {
        if (input == null) return "";
        
        // Remove/encode dangerous characters
        StringBuilder sanitized = new StringBuilder();
        for (char c : input.toCharArray()) {
            switch (c) {
                case '<':
                    sanitized.append("&lt;");
                    break;
                case '>':
                    sanitized.append("&gt;");
                    break;
                case '&':
                    sanitized.append("&amp;");
                    break;
                case '"':
                    sanitized.append("&quot;");
                    break;
                case '\\'':
                    sanitized.append("&#x27;");
                    break;
                case '/':
                    sanitized.append("&#x2F;");
                    break;
                default:
                    if (c >= 32 && c <= 126) {
                        sanitized.append(c);
                    }
                    break;
            }
        }
        return sanitized.toString();
    }
    
    public static void main(String[] args) {
        String[] testCases = {
            "Alice",
            "Bob<script>alert('xss')</script>",
            "Charlie&Drop Table",
            "",
            "A".repeat(150)
        };
        
        for (int i = 0; i < testCases.length; i++) {
            try {
                ByteArrayInputStream request = new ByteArrayInputStream(testCases[i].getBytes());
                ByteArrayOutputStream response = new ByteArrayOutputStream();
                
                String result = httpHelloPage(request, response);
                System.out.println("Test " + (i + 1) + ": " + result);
            } catch (IOException e) {
                System.err.println("Test " + (i + 1) + " failed: " + e.getMessage());
            }
        }
    }
}
